Sources/carton/Model/Template.swift

```diff
@@ -0,0 +1,185 @@
+  let version: Version
+  
+  enum Version: CustomStringConvertible {
+    case from(String)
+    case branch(String)
+    var description: String {
+      switch self {
+      case let .from(min):
+        return #"from: "\#(min)""#
+      case let .branch(branch):
+        return #".branch("\#(branch)")"#
+      }
+    }
+  }
+  
+  var description: String {
+    #".package(name: "\#(name)", url: "\#(url)", \#(version))"#
+  }
+}
+
+struct TargetDependency: CustomStringConvertible {
+  let name: String
+  let package: String
+  var description: String {
+    #".product(name: "\#(name)", package: "\#(package)")"#
+  }
+}
+
+extension Template {
+  static func createPackage(type: PackageType,
+                            fileSystem: FileSystem,
+                            project: Project,
+                            _ terminal: TerminalController) throws {
+    try Toolchain(fileSystem, terminal)
+      .packageInit(name: project.name, type: type)
+  }
+  static func createManifest(fileSystem: FileSystem,
+                             project: Project,
+                             dependencies: [PackageDependency] = [],
+                             targetDepencencies: [TargetDependency] = [],
+                             _ terminal: TerminalController) throws {
+    try fileSystem.writeFileContents(project.path.appending(component: "Package.swift")) {
+      """
+      // swift-tools-version:5.3
+      import PackageDescription
+      let package = Package(
+          name: "\(project.name)",
+          platforms: [.macOS(.v10_15)],
+          products: [
+              .executable(name: "\(project.name)", targets: ["\(project.name)"])
+          ],
+          dependencies: [
+              \(dependencies.map(\.description).joined(separator: ",\n"))
+          ],
+          targets: [
+              .target(
+                  name: "HelloWorld",
+                  dependencies: [
+                      \(targetDepencencies.map(\.description).joined(separator: ",\n"))
+                  ]),
+              .testTarget(
+                  name: "\(project.name)Tests",
+                  dependencies: ["\(project.name)"]),
+          ]
+      )
+      """
+        .write(to: $0)
+    }
+  }
+}
+
+// MARK: Templates
+extension Templates {
+  struct Basic : Template {
+    static let description: String = "A simple SwiftWasm project."
+    
+    static func create(on fileSystem: FileSystem,
+                       project: Project,
+                       _ terminal: TerminalController) throws {
+      try fileSystem.changeCurrentWorkingDirectory(to: project.path)
+      try createPackage(type: .executable,
+                        fileSystem: fileSystem,
+                        project: project,
+                        terminal)
+      try createManifest(fileSystem: fileSystem,
+                         project: project,
+                         terminal)
+    }
+  }
+}
+
+extension Templates {
+  struct Tokamak : Template {
```

Comment:
**nitpick**: would you be able to run `swiftformat .` in the project or pre-commit installed as described in [the "Coding Style" section](https://github.com/swiftwasm/carton#coding-style) of `README.md`?

```suggestion
  struct Tokamak: Template {
```